File: core/src/nl/t64/game/rpg/screens/questlog/SummaryTable.java
package nl.t64.game.rpg.screens.questlog;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.utils.Align;
import nl.t64.game.rpg.Utils;
import nl.t64.game.rpg.components.quest.QuestGraph;
import nl.t64.game.rpg.constants.Constant;


class SummaryTable {

    private static final String TEXT_FONT = "fonts/spectral_extra_bold_20.ttf";
    private static final int TEXT_SIZE = 20;
    private static final float WIDTH = -102f;
    private static final float PAD = 20f;

    final Table container;
    private final Label summary;

    SummaryTable() {
        this.summary = this.createLabel();
        this.container = this.createContainer();
    }

    void populateSummary(QuestGraph quest) {
        summary.setText(quest.getSummary());
        container.getCells().peek().setActor(Utils.getFaceImage(quest.getCharacter()));
    }

    private Label createLabel() {
        var font = Utils.getResourceManager().getTrueTypeAsset(TEXT_FONT, TEXT_SIZE);
        var labelStyle = new Label.LabelStyle(font, Color.BLACK);
        var label = new Label("", labelStyle);
        label.setWrap(true);
        return label;
    }

    private Table createContainer() {
        var table = new Table();
        table.defaults().align(Align.topLeft);
        table.columnDefaults(0).width((Gdx.graphics.getWidth() / 2f) + WIDTH - Constant.FACE_SIZE - PAD).pad(PAD);
        table.columnDefaults(1).size(Constant.FACE_SIZE);
        table.setBackground(Utils.createTopBorder());

        table.add(summary);
        table.add();
        return table;
    }

}

File: core/src/nl/t64/game/rpg/screens/questlog/SummaryTable.kt
package nl.t64.game.rpg.screens.questlog

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Label.LabelStyle
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.utils.Align
import nl.t64.game.rpg.Utils
import nl.t64.game.rpg.Utils.resourceManager
import nl.t64.game.rpg.components.quest.QuestGraph
import nl.t64.game.rpg.constants.Constant


private const val TEXT_FONT = "fonts/spectral_extra_bold_20.ttf"
private const val TEXT_SIZE = 20
private const val WIDTH = -102f
private const val PAD = 20f

internal class SummaryTable {

    private val summary: Label = createLabel()
    val container: Table = createContainer()

    fun populateSummary(quest: QuestGraph) {
        summary.setText(quest.summary)
        container.cells.peek().setActor<Image>(Utils.getFaceImage(quest.entityId))
    }

    private fun createLabel(): Label {
        val font = resourceManager.getTrueTypeAsset(TEXT_FONT, TEXT_SIZE)
        val labelStyle = LabelStyle(font, Color.BLACK)
        return Label("", labelStyle).apply {
            wrap = true
        }
    }

    private fun createContainer(): Table {
        return Table().apply {
            defaults().align(Align.topLeft)
            columnDefaults(0).width(Gdx.graphics.width / 2f + WIDTH - Constant.FACE_SIZE - PAD).pad(PAD)
            columnDefaults(1).size(Constant.FACE_SIZE)
            background = Utils.createTopBorder()
            add(summary)
            add()
        }
    }

}
